src/Evaluators/reduced_evaluator.jl

```diff
@@ -119,14 +156,21 @@ initial(nlp::ReducedSpaceEvaluator) = initial(nlp.model, Control())
 bounds(nlp::ReducedSpaceEvaluator, ::Variables) = nlp.u_min, nlp.u_max
 bounds(nlp::ReducedSpaceEvaluator, ::Constraints) = nlp.g_min, nlp.g_max
 
-function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0)
+function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0, device = nothing)
     x₀ = nlp.x
     jac_x = nlp.autodiff.Jgₓ
     # Transfer x, u, p into the network cache
     transfer!(nlp.model, nlp.buffer, nlp.x, u, nlp.p)
     # Get corresponding point on the manifold
+    if !(device === nothing)
```

Comment:
I would rather use multiple dispatch there, to avoid this `if` statement

Reply:
Done.